spec/services/saml_request_validator_spec.rb

```diff
@@ -96,6 +96,42 @@
           **extra,
         )
       end
+
+      context 'when identity proofing is requested' do
+        let(:authn_context) { [Saml::Idp::Constants::IAL_VERIFIED_ACR] }
+
+        it 'returns FormResponse with success: false' do
+          expect(response.to_h).to eq(
+            success: false,
+            error_details: { service_provider: { unauthorized_service_provider: true } },
```

Comment:
I am diving into the weeds and think this is out of scope, but it took me a bit to realize that this bit on line 90:

```ruby
let(:sp) { ServiceProvider.find_by(issuer: 'foo') }
```

returns nil because it finds nothing. Not knowing this section of code well, I guessed it was a ServiceProvider which existed, but hadn't been "authorized", whatever that might mean.

Would it be more expressive if it was just `let(:sp) { nil }`? Or I guess the point is that _something_ was passed for an issuer, but it didn't match a real SP?

This PR is a simple bugfix + adding a regression test, so please don't let me derail it unless this sparks something on your end.

Reply:
updated in https://github.com/18F/identity-idp/pull/11946/commits/1545660a56573b77e3025eaa400aded31a5c3314 -- i think there is some value to having it mirror the actual thing being passed in, but not at the expense of being clear to the folks reading the tests